imod/protocols/protocol_tomoPreprocess.py

```diff
@@ -341,36 +184,36 @@ def preprocessStep(self, tsId, runNewstack, binning):
 
     def generateOutputStep(self, tsId, runNewstack, binning):
         tomo = self.tomoDict[tsId]
-        if tsId in self._failedTomos:
-            self.createOutputFailedSet(tomo)
-        else:
-            output = self.getOutputSetOfTomograms(self.getInputSet(), binning)
-            newTomogram = Tomogram()
-            newTomogram.copyInfo(tomo)
-
-            if not runNewstack and binning == 1:
-                newTomogram.setLocation(tomo.getFileName())
+        with self._lock:
+            if tsId in self._failedItems:
+                self.createOutputFailedSet(tomo)
             else:
-                newTomogram.setLocation(self.getExtraOutFile(tsId, ext=MRC_EXT))
+                output = self.getOutputSetOfTomograms(self.getInputSet(pointer=True), binning)
+                newTomogram = Tomogram()
+                newTomogram.copyInfo(tomo)
 
-            if binning > 1:
-                newTomogram.setSamplingRate(tomo.getSamplingRate() * binning)
-                # Fix the mrc tomogram
-                newTomogram.fixMRCVolume(setSamplingRate=True)
-                # Set default tomogram origin
-                newTomogram.setOrigin(newOrigin=None)
+                if not runNewstack and binning == 1:
+                    newTomogram.setLocation(tomo.getFileName())
+                else:
+                    newTomogram.setLocation(self.getExtraOutFile(tsId, ext=MRC_EXT))
 
-            if self.oddEvenFlag:
-                halfMapsList = [self.getExtraOutFile(tsId, suffix=ODD, ext=MRC_EXT),
-                                self.getExtraOutFile(tsId, suffix=EVEN, ext=MRC_EXT)]
-                newTomogram.setHalfMaps(halfMapsList)
+                if binning > 1:
+                    newTomogram.setSamplingRate(tomo.getSamplingRate() * binning)
+                    # Fix the mrc tomogram
+                    newTomogram.fixMRCVolume(setSamplingRate=True)
+                    # Set default tomogram origin
+                    newTomogram.setOrigin(newOrigin=None)
 
-            output.append(newTomogram)
-            output.updateDim()
-            output.update(newTomogram)
+                if self.oddEvenFlag:
+                    halfMapsList = [self.getExtraOutFile(tsId, suffix=ODD, ext=MRC_EXT),
+                                    self.getExtraOutFile(tsId, suffix=EVEN, ext=MRC_EXT)]
+                    newTomogram.setHalfMaps(halfMapsList)
 
-            output.write()
-            self._store(output)
+                output.append(newTomogram)
+                output.updateDim()
+                output.update(newTomogram)
+                output.write()
+                self._store(output)
```

Comment:
I can see you put back these two:
output.write()
self._store(output)
Is there a reason? Protocol works without these redundant calls. Write and store are called in the closeOutputSet step for all outputs